src/main/java/sparksoniq/spark/iterator/flowr/CountClauseSparkIterator.java

```diff
@@ -1,33 +1,86 @@
 package sparksoniq.spark.iterator.flowr;
 
 import org.apache.spark.api.java.JavaRDD;
-import sparksoniq.jsoniq.compiler.translator.expr.flowr.FLWOR_CLAUSES;
+import sparksoniq.exceptions.IteratorFlowException;
+import sparksoniq.exceptions.SparksoniqRuntimeException;
+import sparksoniq.jsoniq.item.IntegerItem;
+import sparksoniq.jsoniq.item.Item;
 import sparksoniq.jsoniq.runtime.iterator.RuntimeIterator;
 import sparksoniq.jsoniq.runtime.iterator.primary.VariableReferenceIterator;
 import sparksoniq.jsoniq.runtime.metadata.IteratorMetadata;
 import sparksoniq.jsoniq.runtime.tupleiterator.RuntimeTupleIterator;
+import sparksoniq.jsoniq.runtime.tupleiterator.SparkRuntimeTupleIterator;
 import sparksoniq.jsoniq.tuple.FlworTuple;
 import sparksoniq.semantics.DynamicContext;
 import sparksoniq.spark.closures.CountClauseClosure;
-import sparksoniq.spark.iterator.flowr.base.FlowrClauseSparkIterator;
 
-import java.util.Collections;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CountClauseSparkIterator extends SparkRuntimeTupleIterator {
+    private VariableReferenceIterator _variableReference;
```

Comment:
I think all you need is a String with the name of the variable. This simplifies the signatures.